frontend/src/Settings/index.jsx

```diff
@@ -1,14 +1,130 @@
+
+import InputField from './InputField';
+import SubmitButton from './SubmitButton';
+
+const styles = StyleSheet.create({
+  titleText: {
+    margin: 20,
+    fontSize: 36,
+    color: Colors.whiteColor,
+    fontWeight: 'bold',
+  },
+  labelText: {
+    width: 280,
+    fontSize: 16,
+    color: Colors.whiteColor,
+  },
+  buttonRow: {
+    width: 280,
+    height: 60,
+    paddingTop: 6,
+    flexDirection: 'row-reverse',
+    alignItems: 'flex-start',
+  },
+  container: {
+    flex: 1,
+    paddingTop: 40,
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  background: {
+    width: '100%',
+    height: '100%',
+  },
+  footer: {
+    flex: 1,
+    justifyContent: 'flex-end',
+    marginBottom: 60,
+  },
+  logoutButton: {
+    width: 280,
+  },
+});
 
 function Settings() {
+  const [emailInput, setEmailInput] = useState('');
+  const [usernameInput, setUsernameInput] = useState('');
+  const [kingdomNameInput, setKingdomNameInput] = useState('');
+  const [passwordInput, setPasswordInput] = useState('');
+
+  function handleSubmit() {
+    const settings = {};
+    let message = '';
+    if (emailInput) {
+      settings.email = emailInput;
+      message += 'Your email has changed.\n';
+    }
+    if (usernameInput) {
+      settings.username = usernameInput;
+      message += 'Your username has changed.\n';
+    }
+    if (kingdomNameInput) {
+      settings.kingdom = kingdomNameInput;
+      message += 'Your kingdom\'s name has changed.\n';
+    }
+    if (passwordInput) {
+      settings.password = passwordInput;
+      message += 'Your password has changed.\n';
+    }
+    Alert.alert('Settings', message);
+  }
+
+  const submitButtonIsDisabled = emailInput === '' && usernameInput === '' && kingdomNameInput === '' && passwordInput === '';
+
   return (
-    <View style={[styles.screenStyle,
-      { flex: 1, justifyContent: 'center', alignItems: 'center' }]}
+    <ImageBackground
+      style={styles.background}
+      resizeMode="cover"
+      source={background}
     >
-      <Text>Mocked Settings</Text>
-    </View>
+      <View style={styles.container}>
+        <Text style={styles.titleText}>Settings</Text>
+
+        <Text style={styles.labelText}>email</Text>
+        <InputField
+          placeholder={useSelector((state) => state.settings.email)}
+          value={emailInput}
+          onChangeText={(value) => setEmailInput(value)}
+        />
+
+        <Text style={styles.labelText}>username</Text>
+        <InputField
+          placeholder={useSelector((state) => state.settings.username)}
+          value={usernameInput}
+          onChangeText={(value) => setUsernameInput(value)}
+        />
+
+        <Text style={styles.labelText}>Enter your new Kingdom Name here</Text>
+        <InputField
+          placeholder={useSelector((state) => state.settings.kingdomName)}
+          value={kingdomNameInput}
+          onChangeText={(value) => setKingdomNameInput(value)}
```

Comment:
before saving the new password, please check if it's more than 8 characters long, during registration we are checking it